### PR TITLE
Document how to run local module tests with dev versions of nf-core and/or Nextflow

### PR DESCRIPTION
**Pull request by erikrikarddaniel**
The title says it all, I hope.

### CODE REVIEW

File: markdown/developers/tutorials/dsl2_modules_tutorial.md [hunk 1]
```diff
@@ -420,6 +420,17 @@ PROFILE=conda pytest --tag fgbio_bamtofastq --symlink --keep-workflow-wd
 
 Hopefully everything runs smoothly, and we are then ready to open a pull request, and contribute to the nf-core community.
 
+If you don't have a `pytest` command in your nf-core environment, or you're notified there's no `--symlinks` option, try a later version of nf-core.
```
Comment (drpatelh) [inline markdown/developers/tutorials/dsl2_modules_tutorial.md:423]:
```suggestion
To save you having to install `pytest-workflow` separately it was added as a dependency for nf-core/tools (`>= 1.13.2`). However, if you find that you don't have a `pytest` command in your nf-core environment, or you're notified there's no `--symlinks` option, you could try and install a later version of nf-core/tools to see if that works instead.
```

File: markdown/developers/tutorials/dsl2_modules_tutorial.md [hunk 2]
```diff
@@ -420,6 +420,17 @@ PROFILE=conda pytest --tag fgbio_bamtofastq --symlink --keep-workflow-wd
...
+
+In case there are errors, check if the Nextflow version you used was too old.
+You can see this in `log.err` in the directory where output from the test was run (recorded in the output from the command), e.g.: `Nextflow version 20.10.0 does not match workflow required version: >=20.11.0-edge`.
+To run the test with a later version of Nextflow, in this case 21.04.0-edge (see [Nextflow's release page](https://github.com/nextflow-io/nextflow/releases)):
```
Comment (drpatelh) [inline markdown/developers/tutorials/dsl2_modules_tutorial.md:427]:
```suggestion
The minimum Nextflow version required to run the tests can be found in [this `nextflow.config` file](https://github.com/nf-core/modules/blob/d63ff4ba1b08cd0dc05c375efa69885297de7507/tests/config/nextflow.config#L28) in the nf-core/modules repository. If the version of Nextflow you are using is older than the version specified there you may get an error such as `Nextflow version 20.10.0 does not match workflow required version: >=20.11.0-edge`. The error will be reported in `log.err` in the directory where the outputs from the tests were generated. See the Nextflow [releases](https://github.com/nextflow-io/nextflow/releases) and [installation](https://www.nextflow.io/docs/latest/getstarted.html#installation) pages to install a later version.
```

**Review by drpatelh — COMMENTED** [review]
Thanks @erikrikarddaniel !! Re-worded stuff a little and pointed to where we can get some of this information on nf-core/modules.

File: .gitignore [hunk 3]
```diff
@@ -27,3 +27,4 @@ tmp
 www
 .user.ini
 php.ini
+.*.sw?
```
[the remark this answers is not in the document]
Comment (erikrikarddaniel) [reply .gitignore:30]: Vim swap files, typically .file.swp.

[the remark this answers is not in the document]
Comment (mashehu) [reply .gitignore:30]: we are not ignoring any other editor specific files, so I wouldn't put it there. But maybe I am too strict there. What does the rest of @nf-core/core think?

[the remark this answers is not in the document]
Comment (ewels) [reply .gitignore:30]: I'm happy with this - tend not to be too strict with `.gitignore`

Note that this won't match `.swp` though? Just `.s` and `.sw` unless gitgnore files work differently to how I expect. (Which may well be the case actually.. 🤔 )

[the remark this answers is not in the document]
Comment (ewels) [reply .gitignore:30]: Yeah maybe it's glob not regex so probs fine..

**Review by ewels — APPROVED** [review]
LGTM

**Pull request merged**